dll/opengl/opengl32/glapi_x86.s

```diff
@@ -7,19 +7,47 @@
 
 /* X86 opengl API entry points, fast forward to the current thread's dispatch table */
 #include <asm.inc>
+#include <ks386.inc>
+
+.data
+ASSUME nothing
+
+.align 16
 
 .code
 
+#ifdef OPENG32_USE_TLS
 EXTERN _OglTlsIndex:DWORD
 EXTERN _TlsGetValue@4:PROC
+#endif
 
 MACRO(USE_GL_FUNC, name, offset, stack)
 PUBLIC _gl&name&@&stack
 .PROC _gl&name&@&stack
+    LOCAL fast_ret
+
+#ifdef OPENG32_USE_TLS
     push _OglTlsIndex
     call _TlsGetValue@4
+    /* If we don't have a thread data, this is a nop */
+    test eax, eax
+    jz short fast_ret
+    /* Get the GL table */
     mov eax, [eax]
+#else
+    /* Get the TEB */
+    mov eax, fs:[TEB_SELF]
+    /* Get the GL table */
+    mov eax, [eax + HEX(BE8)]
```

Comment:
Would be nice to create a define for this magic constant as well?

Reply:
Yup.